### PR TITLE
feature: add markdown formatter

### PR DESCRIPTION
**Pull request by tancnle**
## What

This PR adds a markdown formatter that allows users to format all markdown files.

## Why

We would like to have a consistent format of markdown files across our repository. Having automation around this will cut down review effort dramatically, especially comments around markdown styles.

## How

- Implement node and npm packages
- Add [`prettier`](https://prettier.io/) formatter
- Add VSCode plugin to allow formatting on save
- Add [`editorconfig`](https://editorconfig.org/) to ensure consistent EOL and indentation rules
- Add development instructions on `CONTRIBUTING.md`
- Integrate CI to automatically check for markdown style issues

### Usages

- Via VSCode [plugin](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode), formatting can be done via manually triggering the command or formatting on save (see usage section in the plugin [page](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode))
- Via command line, see the `CONTRIBUTING.md` file in this changeset

If the PR includes changes containing markdown style issues, the CI build will fail and merging is blocked.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Contributing
+
+You can help to improve the Thoth Tech company handbook by sending pull requests
+to this repository. Thank you for your interest and help!
```
Comment (ghost) [inline CONTRIBUTING.md:4]: As in all .md files are sent to this repo and then will be made _pretty_?

**Review by ghost — APPROVED** [review]
Looks Good, thanks for the demo

File: CONTRIBUTING.md (hunk 1 above)
Comment (ghost) [reply CONTRIBUTING.md:4]: Thanks to @tancnle for the explanation and demo, was able to answer my question:
If identified as problematic (aka. not _pretty_) the changes will not be allowed to merge/review. A new commit must be created to fix the formatting issues.

Comment (tancnle) [reply CONTRIBUTING.md:4]: To make `.md` files pretty, the author has to run the formatter locally.

1. Via the VS Code plugins
2. Via the command line `npm run format`

**Pull request merged**